watchevents/generic.go

```diff
@@ -0,0 +1,121 @@
+package watchevents
+
+import (
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/rancher/go-rancher/v2"
+	"github.com/rancher/kubernetes-agent/kubernetesclient"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/apis/apps/v1beta1"
+	"k8s.io/client-go/pkg/apis/autoscaling"
+	batchv1 "k8s.io/client-go/pkg/apis/batch/v1"
+	extv1beta1 "k8s.io/client-go/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+)
+
+type genericHandler struct {
+	rClient          *client.RancherClient
+	kClient          *kubernetesclient.Client
+	genericWatchChan []chan struct{}
+}
+
+func NewGenericHandler(rClient *client.RancherClient, kClient *kubernetesclient.Client) *genericHandler {
+	genHandler := &genericHandler{
+		rClient: rClient,
+		kClient: kClient,
+	}
+	return genHandler
+}
+
+func (g *genericHandler) startGenericWatch(resourceName string) chan struct{} {
+	runtimeObj, restClient := g.getRuntimeObject(resourceName)
+	watchlist := cache.NewListWatchFromClient(restClient, resourceName, "", fields.Everything())
+	_, controller := cache.NewInformer(
+		watchlist,
+		runtimeObj,
+		time.Second*0,
```

Comment:
@alena1108 Should any of the watches set a resync period?

Reply:
@galal-hussein looks like we can drop generic events handling. It was used for old UI only, we do not need it anymore.